hive/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -83,6 +86,18 @@ protected HiveTableOperations(Configuration conf, HiveClientPool metaClients, St
     this.tableName = table;
     this.lockAcquireTimeout =
         conf.getLong(HIVE_ACQUIRE_LOCK_STATE_TIMEOUT_MS, HIVE_ACQUIRE_LOCK_STATE_TIMEOUT_MS_DEFAULT);
+    try {
+      this.alterTableMethod = HiveMetaStoreClient.class.getMethod("alter_table_with_environmentContext",
+          String.class, String.class, Table.class, EnvironmentContext.class);
+    } catch (NoSuchMethodException e) {
+      try {
+        this.alterTableMethod = HiveMetaStoreClient.class.getMethod("alter_table",
+          String.class, String.class, Table.class, EnvironmentContext.class);
+      } catch (NoSuchMethodException inner) {
+        throw new RuntimeMetaException(inner, "Fail to find method alter_table_with_environmentContext and " +
+            "alter_table with right signature");
+      }
```

Comment:
Use reflection to handle the API difference between Hive 1.2.1 and 2.3.6.

Reply:
Can you update this to use the reflection utilities from `iceberg-common`? Those handle exceptions and use a builder pattern so it's concise and easy to maintain across versions:

```java
  private static final DynMethods.UnboundMethod ALTER_TABLE = DynMethods
      .builder("alter_table")
      .impl(HiveMetaStoreClient.class, "alter_table_with_environmentContext",
          String.class, String.class, Table.class, EnvironmentContext.class)
      .impl(HiveMetaStoreClient.class, "alter_table",
          String.class, String.class, Table.class, EnvironmentContext.class)
      .build();
```